source/Calamari/Commands/FindPackageCommand.cs

```diff
@@ -78,7 +78,8 @@ public override int Execute(string[] commandLineArguments)
             //Exact package found, so we need to register use and lock it.
             // This command can't use the PackageJournalCommandDecorator because we don't lock on partial finds, which the decorator would include.  We don't lock on partials because there may be too many packages, blocking retention later,
             //  and we will lock them on apply delta anyway.
-            if (variables.IsPackageRetentionEnabled()) packageJournal.RegisterPackageUse(variables);
+            var tentacleHome = Environment.GetEnvironmentVariable("TentacleHome");
+            if (variables.IsPackageRetentionEnabled() && tentacleHome != null) packageJournal.RegisterPackageUse(variables);
```

Comment:
Do we also need to add the `tentacleHome` check to `PackageJournalCommandDecorator`?

Reply:
I think that sounds better

Reply:
I went with a public method which includes `variables.IsPackageRetentionEnabled() && tentacleHome != null`. The checks are still in the same places (`PackageJournalCommandDecorator` and `FindPackageCommand`) but they're not concerned with what the check is doing anymore.

Reply:
I'm a bit concerned about the commands having to worry about whether or not retention is enabled - We can pass the variables to the Journal (even though that creates coupling between the journal and the variables collection coming from server), and set a "IsRetentionEnabled" property on the journal, and check that at the start of the registerPackageUse() etc variables.

This also means the journal knows about the package info (because of the variables collection), and we can modify the RegisterPackageUse() methods to not take in any parameters - though that part should be in another PR....

Reply:
I agree with you on the commands still having to check for retention being enabled. Looking at it again I'm not completely happy with how it is. Adding the variables to the journal sounds like a nicer alternative. For now I can make those changes and just leave the parameters of `RegisterPackageUse` how they are for the time being.